x-pack/plugins/security_solution/public/common/containers/source/index.tsx

```diff
@@ -111,27 +129,37 @@ export const useFetchIndex = (
         try {
           setState({ ...state, loading: true });
           abortCtrl.current = new AbortController();
-          const dv = await data.dataViews.create({ title: iNames.join(','), allowNoIndex: true });
-          const dataView = dv.toSpec();
-          const { browserFields } = getDataViewStateFromIndexFields(iNames, dataView.fields);
+          let dataView: DataViewSpec;
+          if (!isIndexNameArray) {
+            const fulldv = await data.dataViews.get(indexNamesOrDvId);
+            dataView = fulldv.toSpec();
+            previousIndexesNameOrId.current = dataView?.id;
+          } else {
+            const dv = await data.dataViews.create({ title: iNames.join(','), allowNoIndex: true });
+            // console.error(dv.fields);
+            dataView = dv.toSpec();
+          }
 
-          previousIndexesName.current = dv.getIndexPattern().split(',');
+          const { browserFields } = getDataViewStateFromIndexFields(iNames, dataView.fields);
+          previousIndexesNameOrId.current = dataView?.title?.split(',');
```

Comment:
should not this `previousIndexesNameOrId.current = dataView?.title?.split(',');` be within previous `else` branch? it will always override line 136